File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 7/10/2025, 9:00:01 AM
class Solution {
    public boolean isPalindrome(int x) {
        int temp = x;
        int reversed = 0;
        if( x < 0){
            return false;
        }
        while( x != 0){
            int digit = x % 10;
            x /= 10;
            reversed = reversed * 10 + digit;
        }
        if( temp == reversed){
            return true;
        }
        return false;
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.c
// Last updated: 7/10/2025, 9:00:07 AM
bool isPalindrome(int x) {
 if (x < 0) {
        return false;
    }
    int a = x;
    long b = 0;
    while (x > 0) {
        int digit = x % 10;
        b = b * 10 + digit;
        x /= 10;
    }
    return a == b;
}